contrib/win32/win32compat/ssh-agent/keyagent-request.c

```diff
@@ -268,6 +365,66 @@ process_sign_request(struct sshbuf* request, struct sshbuf* response, struct age
 	int r, request_invalid = 0, success = 0;
 	struct sshkey *key = NULL;
 
+#ifdef ENABLE_PKCS11
+	int i, count = 0, index = 0;;
+	wchar_t sub_name[MAX_KEY_LENGTH];
+	DWORD sub_name_len = MAX_KEY_LENGTH;
+	DWORD pin_len, epin_len, provider_len;
+	char *pin = NULL, *epin = NULL, *provider = NULL;
+	HKEY root = 0, sub = 0, user_root = 0;
+	struct sshkey **keys = NULL;
+
+	pkcs11_init(0);
+
+	if (get_user_root(con, &user_root) != 0 ||
+		RegOpenKeyExW(user_root, SSH_PKCS11_PROVIDERS_ROOT, 0, STANDARD_RIGHTS_READ | KEY_ENUMERATE_SUB_KEYS | KEY_WOW64_64KEY, &root) != 0) {
+		success = 1;
+		goto done;
+	}
+
+	while (1) {
+		sub_name_len = MAX_KEY_LENGTH;
+		if (sub) {
+			RegCloseKey(sub);
+			sub = NULL;
+		}
+		if (RegEnumKeyExW(root, index++, sub_name, &sub_name_len, NULL, NULL, NULL, NULL) == 0) {
+			if (RegOpenKeyExW(root, sub_name, 0, KEY_QUERY_VALUE | KEY_WOW64_64KEY, &sub) == 0 &&
```

Comment:
Are we missing a `RegCloseKey(sub)` here? (when we loop)

Reply:
We are calling `RegCloseKey(sub)` at the beginning of each iteration and one last time before we return.